backend/app.py

```diff
@@ -76,3 +76,130 @@ def _streamer(last_event_id: int | None = None):
 
     last_event_id = request.headers.get("Last-Event-ID")
     return _streamer(last_event_id), 200, {"Content-Type": "text/event-stream"}
+
+
+@app.route("/tracker/v4", methods=["GET"])
+def pq_caching_tracker():
+    """Uses PostgreSQL for caching stock prices to improve performance"""
+
+    def _streamer(last_event_id: int | None = None):
+        with app.app_context():
+            connection = db.engine.raw_connection()
+            cursor = connection.cursor()
+            
+            # Create a temporary table for caching if it doesn't exist
+            cursor.execute("""
+                CREATE TABLE IF NOT EXISTS stock_price_cache (
+                    id SERIAL PRIMARY KEY,
+                    price_id INTEGER REFERENCES stock_price(id),
+                    payload JSONB,
+                    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+                    accessed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+                )
+            """)
```

Comment:
So are we trying to dynamically create a new table every time someone new hits the API?